File: Data Structures & Algorithms/house-robber/submission-2.java
class Solution {
    public int rob(int[] nums) {
        int n = nums.length;
        int []c = new int [n+1];
        if (n == 0) return 0;
        if (n == 1) return nums[0];

        c[0] = nums[0];
        c[1] = Math.max(nums[0],nums[1]);

        for (int i = 2; i < n; i++){
            int IN = nums[i] + c[i-2];
            int OUT = c[i-1];
            c[i] = Math.max(IN,OUT);
        }

        return c[n-1];
    }//[0,1,1,]
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        best = 0

        for right in range(len(prices)):
            if prices[right] < prices[left]:
                left = right
            var = prices[right] - prices[left]

            best = max(var, best)
        return best
        